### PR TITLE
feat(server): WebsocketServer implementation and tests

### PR DESCRIPTION
**Pull request by Loonride**
- [ ] This is a **bugfix**
- [x] This is a **feature**
- [ ] This is a **code refactor**
- [ ] This is a **test update**
- [ ] This is a **docs update**
- [ ] This is a **metadata update**

### For Bugs and Features; did you add new tests?

Yes, but this cannot be tested in e2e without a native WebSockets client implementation.

### Motivation / Use-Case

Add `WebsocketServer` implementation, trying to mirror the features of `SockJSServer`.

### Additional Info

Should I keep the `this.server.log.error(err.message);` line? I will add a test for it if I keep it, I am just trying to mirror the `SockJSServer` implementation.

### CODE REVIEW

File: test/server/servers/WebsocketServer.test.js [hunk 1]
```diff
@@ -0,0 +1,67 @@
+'use strict';
+
+const http = require('http');
+const express = require('express');
+const ws = require('ws');
+const WebsocketServer = require('../../../lib/servers/WebsocketServer');
+const port = require('../../ports-map').WebsocketServer;
+
+describe('SockJSServer', () => {
```
Comment (hiroppy) [inline test/server/servers/WebsocketServer.test.js]: `WebsocketServer`

Comment (Loonride) [reply test/server/servers/WebsocketServer.test.js]: good catch

**Comment by hiroppy** [comment]
need to rebase

**Comment by Loonride** [comment]
@evilebottnawi Ready for review.

Yes, after this I need to add `'ws'` for `clientMode` and `serverMode`. Then once those are both implemented, I will add some e2e tests to confirm that they work together.

**Comment by evilebottnawi** [comment]
/cc @Loonride ready for 
> Yes, after this I need to add 'ws' for clientMode and serverMode. Then once those are both implemented, I will add some e2e tests to confirm that they work together.

**Pull request merged**